jupyterhub/handlers/base.py

```diff
@@ -248,6 +248,17 @@ def redirect_to_server(self):
     def authenticate_prometheus(self):
         return self.settings.get('authenticate_prometheus', True)
 
+    async def get_current_user_named_server_limit(self):
+        """
+        Return named server limit for current user.
+        """
+        named_server_limit_per_user = self.named_server_limit_per_user
+
+        if callable(named_server_limit_per_user):
+            return await maybe_future(named_server_limit_per_user(self))
+        else:
+            return named_server_limit_per_user
```

Comment:
The `else` here can be removed for simplicity.

```suggestion
      return named_server_limit_per_user
```

Reply:
Done. Thanks!